src/NuGet.Clients/NuGet.CommandLine/Common/NuGetTestMode.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Common
+{
+    // Note: we may delete this logic once we no longer depend on NuGet.Core's PackageServer type,
+    // as it is embedded in NuGet.Protocol.Core.Types.UserAgent.
+    internal static class NuGetTestMode
+    {
+        private const string _testModeEnvironmentVariableName = "NuGetTestModeEnabled";
```

Comment:
Looks like you are using the same environmental variable "NuGetTestModeEnabled" for both command line and functional tests and not clear it after the test runs.

So when manual usage of NuGet is performed after the test run, it would be still considered as "NuGet Test Client". We should probably remove and reset that environmental variable when test run has finished.